memebot/memebot.py

```diff
@@ -173,17 +174,26 @@ async def skip(command_arg: str, channel: discord.TextChannel, player_config: Pl
         )
     return skip
 
+async def queue_display(command_arg: str, channel: discord.TextChannel, player_config: PlayerConfig):
+    if len(player_config.player.playback_queue) == 0:
+        return await channel.send('Nothing in queue. Add a song with !play', delete_after=30)
+    queue = '\n'.join(f'[{idx}] {item.title}' for idx,item in enumerate(player_config.player.playback_queue))
+    return await channel.send(f'Up next:\n{queue}', delete_after=60)
 
 def create_show_queue_executor(player_config_provider: PlayerConfigProvider) -> CommandExecutor:
     @ignore_by_player_config_provider(player_config_provider)
     async def show_queue(command_arg: str, channel: discord.TextChannel, player_config: PlayerConfig):
         '''shows current queue'''
-        if len(player_config.player.playback_queue) == 0:
-            return await channel.send('Nothing in queue. Add a song with !play', delete_after=30)
-        queue = '\n'.join(f'- {item.title}' for item in player_config.player.playback_queue)
-        return await channel.send(f'Up next:\n{queue}', delete_after=60)
+        await queue_display(command_arg, channel, player_config)
     return show_queue
 
+def create_queue_remove_executor(player_config_provider: PlayerConfigProvider) -> CommandExecutor:
+    @ignore_by_player_config_provider(player_config_provider)
+    async def queue_remove(command_arg: str, channel: discord.TextChannel, player_config: PlayerConfig):
+        """allows the removal of a specific song from the queue"""
+        await player_config.player.remove(command_arg)
+        await queue_display(command_arg, channel, player_config)
```

Comment:
[optional] maybe we should also add a success/fail to remove message? If you do add one, maybe pass an string to queue_display for that so we only send one message for the whole transaction

Reply:
OK, will consider strongly unless annoying to do